src/libmongoc/src/mongoc/mongoc-secure-channel.c

```diff
@@ -207,14 +222,67 @@ decode_object (const char *structType,
    return out;
 }
 
+// `generate_key_name` generates a deterministic name for a key of the form: "libmongoc-<SHA256 fingerprint>-<suffix>".
+// Returns NULL on error.
+static LPWSTR
+generate_key_name (LPBYTE data, DWORD len, const char *suffix)
+{
+   bool ok = false;
+   char *hash_hex = NULL;
+   char *key_name = NULL;
+   LPWSTR key_name_wide = NULL;
+
+   BSON_ASSERT_PARAM (data);
+   BSON_ASSERT_PARAM (suffix);
+
+   // Compute a hash of the certificate:
+   {
+      unsigned char hash[32];
+      mongoc_crypto_t crypto;
+      mongoc_crypto_init (&crypto, MONGOC_CRYPTO_ALGORITHM_SHA_256);
+      BSON_ASSERT (mlib_in_range (size_t, len));
+      if (!mongoc_crypto_hash (&crypto, (const unsigned char *) data, (size_t) len, hash)) {
```

Comment:
Can also use `mlib_assert_narrow` to do the check-and-cast inline, and it will print the bad value in case of assertion failure.

```suggestion
      if (!mongoc_crypto_hash (&crypto, 
                               (const unsigned char *) data,
                               mlib_assert_narrow (size_t, len), 
                               hash)) {
```

Reply:
Neat. Thank you for the suggestion.